### PR TITLE
Add support for 2xCAN

### PR DESCRIPTION
**Pull request by haitomatic**
## Summary
Add another instance of CANFD driver resulting in 2 instances now supported

## Impact
CANFD driver

## Testing
Saluki v2 + PX4 UAVCAN flight

### CODE REVIEW

**Review by jlaitine — COMMENTED** [review]
Please check the build error, thanks !

File: arch/risc-v/src/mpfs/Kconfig [hunk 1]
```diff
@@ -511,34 +511,65 @@ config MPFS_COREPWM1_NCHANNELS
...
+	depends on MPFS_HAVE_CANFD
+
+config MPFS_CANFD_BASE1
+	hex "Base address for the CANFD1 instance"
+	default 0x47000000
+	depends on MPFS_CANFD1
 
-config MPFS_CANFD_CLK
-	int "Clock frequency of the CANFD block (Hz)"
+config MPFS_CANFD_CLK1
```
Comment (jlaitine) [inline arch/risc-v/src/mpfs/Kconfig:558]: I think this is always the same as the FPGA clock, so it doesn't need to be defconfig.. It fits better to board.h, but there is already a define for it, see CorePWM for example.

Also note that it was changed to 50000000 in out fpga designs

**Review by jlaitine — COMMENTED** [review]
Would it make sense to make the number of canfd:s configurable the same way as in corespi? For the buffers, just allocate a table with n of blocks, for initialization, do it in loop?

this is not a must requirement, just an idea

**Comment by haitomatic** [comment]
I was thinking about that and thought we have physical limit of only 2 CAN buses on the saluki platform anyway, so I decided to just duplicate the 1st instance. And the icicle board as well as the Pixhawk also got 2. Do we plan to add more CAN buses (more than 2) in the future?

File: arch/risc-v/src/mpfs/Kconfig (hunk 1 above)
Comment (haitomatic) [reply arch/risc-v/src/mpfs/Kconfig]: yes, I rebased and made the change to use peripheral clock config

**Review by jlaitine — APPROVED** [review]

**Pull request merged**